packages/language/src/typesystem/descriptions.ts

```diff
@@ -0,0 +1,1061 @@
+  description: BaseTypeDescription,
+): description is TaskTypeDescription {
+  return description.type === TaskType;
+}
+
+//--- Unknown ---
+const UnknownType = DataType.Unknown;
+type UnknownType = typeof UnknownType;
+
+interface UnknownTypeDescription extends BaseTypeDescription {
+  type: UnknownType;
+}
+
+function createUnknownTypeDescription(): UnknownTypeDescription {
+  return {
+    type: UnknownType,
+    ...createBaseTypeDescription(UnknownType, {}),
+  };
+}
+
+export namespace TypeDescriptions {
+  export type Any =
+    | Area
+    | Arithmetic
+    | File
+    | Format
+    | Label
+    | Locator
+    | Entry
+    | Ordinal
+    | Picture
+    | String
+    | Task
+    | Unknown;
+  export type TypeDescriptionType = Any["type"];
+
+  export const Unknown = createUnknownTypeDescription;
+  export type Unknown = UnknownTypeDescription;
+
+  export const Area = createAreaTypeDescription;
+  export type Area = AreaTypeDescription;
+  export const isArea = isAreaTypeDescription;
+
+  export const Arithmetic = createArithmeticTypeDescription;
+  export type Arithmetic = ArithmeticTypeDescription;
+  export const isArithmetic = isArithmeticTypeDescription;
+
+  export const File = createFileTypeDescription;
+  export type File = FileTypeDescription;
+  export const isFile = isFileTypeDescription;
+
+  export const Format = createFormatTypeDescription;
+  export type Format = FormatTypeDescription;
+  export const isFormat = isFormatTypeDescription;
+
+  export const Label = createLabelTypeDescription;
+  export type Label = LabelTypeDescription;
+  export const isLabel = isLabelTypeDescription;
+
+  export const Locator = createLocatorTypeDescription;
+  export type Locator = LocatorTypeDescription;
+  export const isLocator = isLocatorTypeDescription;
+
+  export const Entry = createEntryTypeDescription;
+  export type Entry = EntryTypeDescription;
+  export const isEntry = isEntryTypeDescription;
+
+  export const Ordinal = createOrdinalTypeDescription;
+  export type Ordinal = OrdinalTypeDescription;
+  export const isOrdinal = isOrdinalTypeDescription;
+
+  export const Picture = createPictureTypeDescription;
+  export type Picture = PictureTypeDescription;
+  export const isPicture = isPictureTypeDescription;
+
+  export const String = createStringTypeDescription;
+  export type String = StringTypeDescription;
+  export const isString = isStringTypeDescription;
+
+  export const Task = createTaskTypeDescription;
+  export type Task = TaskTypeDescription;
+  export const isTask = isTaskTypeDescription;
+
+  /** fake type */
+  export const Boolean = createStringTypeDescription({
+    kind: StringKind.Bit,
+    format: StringFormat.NonVarying,
+    length: 1,
+  });
+  export const isBoolean = (
+    type: TypeDescriptions.Any,
+  ): type is StringTypeDescription =>
+    isString(type) && type.kind === StringKind.Bit && type.length === 1;
+
+  //TODO check default values
+  const DefaultValues: AttributeTypes = {
```

Comment:
There are several places in this file with default values. A opportunity to collect them here.